tsdb/chunks/head_chunks.go

```diff
@@ -226,9 +231,50 @@ func listChunkFiles(dir string) (map[int]string, error) {
 		}
 		res[int(seq)] = filepath.Join(dir, fi.Name())
 	}
+
 	return res, nil
 }
 
+// repairLastChunkFile deletes the last file if it's empty.
+// Because we don't fsync when creating these file, we could end
+// up with an empty file at the end during an abrupt shutdown.
+func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr error) {
+	lastFile := -1
+	for seq := range files {
+		if seq > lastFile {
+			lastFile = seq
+		}
+	}
+
+	if lastFile <= 0 {
+		return files, nil
+	}
+
+	f, err := os.Open(files[lastFile])
+	if err != nil {
+		return files, errors.Wrap(err, "open file during repair")
+	}
+	defer func() {
+		var merr tsdb_errors.MultiError
+		merr.Add(returnErr)
+		merr.Add(f.Close())
+		returnErr = merr.Err()
+	}()
+	info, err := f.Stat()
+	if err != nil {
+		return files, errors.Wrap(err, "file stat during repair")
+	}
+	if info.Size() == 0 {
+		// Corrupt file, hence remove it.
+		if err := os.RemoveAll(files[lastFile]); err != nil {
```

Comment:
May you confirm it's safe removing before `Close()`? To my knowledge it is, but I would like your confirmation too.

Reply:
It works on linux no problem, windows has problem with it I see